src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -839,7 +841,7 @@ public void upgrade_noop(ChocolateyConfiguration config)
                 get_environment_before(config, allowLogging: true);
 
                 var beforeUpgradeAction = new Action<PackageResult>(packageResult => before_package_modify(packageResult, config));
-                var results = perform_source_runner_function(config, r => r.upgrade_run(config, action, beforeUpgradeAction));
+                var results = perform_source_runner_function(config, r => r.upgrade_run(config, action, resetConfigAction, beforeUpgradeAction));

```

Comment:
Any reason you decided to change the `ISourceRunner` interface instead of resetting the configuration inside the `perform_source_runner_function` method?

It sounds a bit wrong to me having to update the interface just to reset the configuration in the `NugetService`

Reply:
It's been long enough since I made this PR that I don't remember exactly, but I think I tried that, and it did not work out.
I'd agree that not changing the interface would be better, so if you can get it to work that would be great.

Reply:
I have replaced the commit with alternative handling (added you as the co-maintainer).

If you could have a test run of the change to verify it still works as you expect, I would appreciate it.

I used the testing steps you have in the PR body as the base of the changes.

If you check it out sometime today, could you reference @corbob to make the review and final merge of the changes?

I have backed up the changes you made and pushed them up to a different branch just in case.

Reply:
Your commit seems to work as well as my previous version did, however, I found another bug that affects both versions, see below comment.